clients/src/main/java/org/apache/kafka/clients/NodeApiVersions.java

```diff
@@ -233,4 +256,8 @@ public ApiVersion apiVersion(ApiKeys apiKey) {
     public Map<ApiKeys, ApiVersion> allSupportedApiVersions() {
         return supportedVersions;
     }
+
+    public Map<String, SupportedVersionRange> supportedFeatures() {
+        return Collections.unmodifiableMap(supportedFeatures);
```

Comment:
It seems like `supportedFeatures` should be immutable and final in any case. There is no use-case for mutating this after the object is constructed.